tools/gengen/util/generator.go

```diff
@@ -0,0 +1,651 @@
+	if err != nil {
+		return address.Undef, address.Undef, err
+	}
+
+	// Note: the owner of the miner is implicit in the msg. The owner will be the `from` address.
+	out, err := g.vm.ApplyGenesisMessage(wAddr, builtin.StoragePowerActorAddr, builtin.MethodsPower.CreateMiner, abi.NewTokenAmount(100000), &power.CreateMinerParams{
+		Worker:     wAddr,
+		Peer:       pid,
+		SectorSize: m.SectorSize,
+	}, &g.chainRand)
+	if err != nil {
+		return address.Undef, address.Undef, err
+	}
+
+	// get miner ID address
+	ret := out.(*power.CreateMinerReturn)
+	return wAddr, ret.IDAddress, nil
+}
+
+func (g *GenesisGenerator) commitDeals(ctx context.Context, addr, mIDAddr address.Address, m *CreateStorageMinerConfig) ([]abi.DealID, error) {
+
+	comms := m.CommittedSectors
+	// Look up worker address's id address
+	stateRoot, err := g.flush(ctx)
+	if err != nil {
+		return nil, err
+	}
+	view := gfcstate.NewView(g.cst, stateRoot)
+	workerID, err := view.InitResolveAddress(ctx, addr)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add 0 balance to escrow and locked table
+	_, err = g.vm.ApplyGenesisMessage(addr, builtin.StorageMarketActorAddr, builtin.MethodsMarket.AddBalance, specsbig.Zero(), &workerID, &g.chainRand)
+	if err != nil {
+		return nil, err
+	}
+	_, err = g.vm.ApplyGenesisMessage(addr, builtin.StorageMarketActorAddr, builtin.MethodsMarket.AddBalance, specsbig.Zero(), &mIDAddr, &g.chainRand)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add all deals to chain in one message
+	params := &market.PublishStorageDealsParams{}
+	for _, comm := range comms {
+		proposal := market.DealProposal{
+			PieceCID:             comm.DealCfg.CommP,
+			PieceSize:            abi.PaddedPieceSize(comm.DealCfg.PieceSize),
+			Client:               workerID,
+			Provider:             mIDAddr,
+			StartEpoch:           0,
+			EndEpoch:             abi.ChainEpoch(comm.DealCfg.EndEpoch), // TODO min and max epoch bounds
+			StoragePricePerEpoch: specsbig.Zero(),
+			ProviderCollateral:   specsbig.Zero(), // collateral should actually be good
+			ClientCollateral:     specsbig.Zero(),
+		}
+		proposalBytes, err := encoding.Encode(proposal)
+		if err != nil {
+			return nil, err
+		}
+		sig, err := crypto.Sign(proposalBytes, g.keys[m.Owner].PrivateKey, crypto.SigTypeBLS)
+		if err != nil {
+			return nil, err
+		}
+
+		params.Deals = append(params.Deals, market.ClientDealProposal{
+			Proposal:        proposal,
+			ClientSignature: sig,
+		})
+	}
+
+	// apply deal builtin.MethodsMarket.PublishStorageDeals
+	out, err := g.vm.ApplyGenesisMessage(addr, builtin.StorageMarketActorAddr, builtin.MethodsMarket.PublishStorageDeals, specsbig.Zero(), params, &g.chainRand)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := out.(*market.PublishStorageDealsReturn)
+
+	return ret.IDs, nil
+}
+
+func (g *GenesisGenerator) setupPost(ctx context.Context, addr, mIDAddr address.Address) error {
+	mAct, found, err := g.stateTree.GetActor(ctx, mIDAddr)
```

Comment:
There is a bit of code duplication in getting and setting actors.  It didn't seem like enough to warrant creating a utility (especially because I don't know of any other places apart from genesis construction this is used) but I will do this if reviewers think duplication rises to that level.